react/features/app-settings/components/AbstractAppSettings.js

```diff
@@ -108,14 +109,48 @@ export class AbstractAppSettings extends Component<Props> {
         });
     }
 
+    _onFinalizeServerURL: () => void;
+
+    /**
+     * Cleans up server URL when the user leaves the field or the screen.
+     * This is required since we want to let the user to freely edit the field
+     * without any limitation at edit time.
+     *
+     * @protected
+     * @returns {void}
+     */
+    _onFinalizeServerURL() {
+        const { serverURL } = this.props._profile;
+
+        if (serverURL) {
+            const urlRegExp = new RegExp('^(\\w+(?:s)?://)?(.+)$');
+            const urlComponents = urlRegExp.exec(serverURL.toLowerCase());
+
+            if (!urlComponents[1] || !urlComponents[1].startsWith('http')) {
+                try {
+                    this._onChangeServerURL(
+                        new URL(`https://${urlComponents[2]}`).toString()
+                    );
+                } catch (error) {
+                    // the user entered an unparseable value we cannot fix,
+                    // so we revert to default.
+                    this._onChangeServerURL('https://zolika.jitsi.net');
```

Comment:
I expect this to be `undefined`, right?

Reply:
Right, sorry, I left the debug value in. But it should be empty string instead.